#157 Profile entry page

deawer234 opened the pull request:
Added profile information to show socials, interests, university and basic info

deawer234 committed:
added basic profile info

deawer234 committed:
added basic profile info

deawer234 committed:
added basic profile info

deawer234 committed:
Merge branch 'issue-157' of https://github.com/esnvutbrno/buena-fiesta into issue-157

deawer234 committed:
profile entry page completed

deawer234 committed:
profile entry page completed

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

deawer234 committed:
#197 fixed instagram username regex

thejoeejoee commented on .enve:
```diff
@@ -0,0 +1,10 @@
+DJANGO_CONFIGURATION=Development
```
should not be included in repository

thejoeejoee commented on issue-157:
```diff
@@ -0,0 +1,45 @@
+  develop[m
```
should not be added

thejoeejoee commented on fiesta/apps/accounts/templatetags/user_profile.py:
```diff
@@ -31,3 +32,12 @@ def compute_profile_fullness(user: User) -> float:
+
+
+@register.simple_tag
+def get_user_interests(user: User | None):
```
Probably not needed -- Django automatically generates [`get_FOO_display`](https://docs.djangoproject.com/en/4.2/ref/models/instances/#django.db.models.Model.get_FOO_display) method for each field with `choices=`, which can be used for getting "human-readable" selected value/s. So `{% for interest in object.user.profile.get_interests_display %}` in template should be fine.


P.S. It's little bit more complicated for ArrayField, but we've our own [custom implementation](https://github.com/esnvutbrno/buena-fiesta/blob/eb582b032c1f4c795ac1ed83fae161bfc19df280/fiesta/apps/utils/models/fields.py#L8-L24).

thejoeejoee commented on fiesta/apps/accounts/templates/accounts/user_profile/detail.html:
```diff
@@ -2,19 +2,131 @@
+      {% with profile=object.user.profile_or_none %}
+        <tr>
+          <td class="font-semibold pr-12">Home University:</td>
+          <td>{{ profile.home_university }}</td>
```
should be defaulted to `home_faculty.university`, currently shows `None`

deawer234 replied:
I havent been able to set the university and faculty at the same time, probably a bug ? Probably will make a separate issue considering it

deawer234 replied:
Actually mentioned in issue #198

thejoeejoee commented on fiesta/apps/accounts/templates/accounts/user_profile/detail.html:
```diff
@@ -2,19 +2,131 @@
+          <td class="font-semibold pr-12">Facebook:</td>
+          <td>
+            <a class="underline text-blue-600 hover:text-blue-800 visited:text-purple-600"
+               href="{{ profile.facebook }}">{{ profile.facebook }}</a>
```
some kind of text truncating should here (probably with tailwind), otherwise overflows the blocks:

<img width="1102" alt="image" src="https://github.com/esnvutbrno/buena-fiesta/assets/6154740/1f89a8be-d098-4ab5-81f8-44ada779c164">

thejoeejoee commented on fiesta/apps/accounts/templates/accounts/user_profile/detail.html:
```diff
@@ -2,19 +2,131 @@
+
+  <div class="bg-gray-100 shadow rounded-box p-4 mb-4 md:box-content">
+    <h1 class="card-title pb-1 mb-3 border-b-2">{% trans "Interests" %}</h1>
+    <div id="id_interests" class="inline-flex">
```
probably `flex-wrap` should be included:

<img width="1183" alt="image" src="https://github.com/esnvutbrno/buena-fiesta/assets/6154740/8115c1b9-6aaf-4975-8163-3481e6e75a8f">

thejoeejoee replied:
`id="id_interests"` is not needed probably

thejoeejoee commented on fiesta/apps/accounts/templates/accounts/user_profile/detail.html:
```diff
@@ -2,19 +2,131 @@
+    <div id="id_interests" class="inline-flex">
+      {% get_user_interests object.user as interests %}
+      {% for interest in interests %}
+        <div class="mr-2">
+          <label class="border divide-gray-100 rounded-full border-gray-400 pt-1 pb-1 pl-2 pr-2">{{ interest }}</label>
+        </div>
```
```suggestion
      <span class="mr-2 border divide-gray-100 rounded-full border-gray-400 pt-1 pb-1 pl-2 pr-2">{{ interest }}</span>
```

should be enough

deawer234 committed:
#197 fixed instagram username regex (#206)

In instagram user name only underscores and punctuations is allowed,
removed minus sign from the regex and tested in onlineregex editor

deawer23 committed:
profile feeback fixes

deawer23 committed:
profile feeback fixes

deawer23 committed:
Merge branch 'issue-157' of https://github.com/esnvutbrno/buena-fiesta into issue-157

deawer23 committed:
deleted not needed function to display interests

deawer234 commented:
pre-commit.ci autofix

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

deawer234 committed:
Fixed uni showing none

deawer234 committed:
deleting annoying things again

petrceska reviewed (approved):
All good

deawer234 committed:
small appereance fixes
